cpp/oneapi/dal/test/engine/csr_table_builder.hpp

```diff
@@ -0,0 +1,175 @@
+            // Remove duplications
+            std::int64_t dup_count = 0;
+            for (std::int32_t i = 1; i < nnz_col_count; ++i) {
+                auto cur_ptr = col_indices_ptr + (fill_count - dup_count);
+                if (cur_ptr[i] == cur_ptr[i - 1]) {
+                    ++dup_count;
+                    // Shift the tail if there is duplication
+                    for (std::int32_t j = i + 1; j < nnz_col_count; ++j) {
+                        cur_ptr[j - 1] = cur_ptr[j];
+                    }
+                }
+            }
+            fill_count += (nnz_col_count - dup_count);
+            // Update row offsets
+            row_offsets_ptr[row_idx + 1] = fill_count + indexing_shift;
+            row_idx++;
+        }
+        if (row_idx < row_count_) {
+            for (std::int32_t i = row_idx; i <= row_count_; ++i) {
+                row_offsets_ptr[i] = nonzero_count + indexing_shift;
+            }
+        }
+    }
+
+#ifdef ONEDAL_DATA_PARALLEL
+    csr_table build_csr_table(device_test_policy& policy) const {
+        auto queue = policy.get_queue();
+        auto nnz_count = data_.get_count();
+        const auto copied_data =
+            dal::array<Float>::empty(queue, nnz_count, sycl::usm::alloc::device);
+        const auto copied_col_indices =
+            dal::array<std::int64_t>::empty(queue, nnz_count, sycl::usm::alloc::device);
+        const auto copied_row_offsets =
+            dal::array<std::int64_t>::empty(queue, row_count_ + 1, sycl::usm::alloc::device);
+        auto data_event = dal::backend::copy_host2usm(queue,
+                                                      copied_data.get_mutable_data(),
+                                                      data_.get_data(),
+                                                      nnz_count);
+
+        auto col_indices_event = dal::backend::copy_host2usm(queue,
+                                                             copied_col_indices.get_mutable_data(),
+                                                             column_indices_.get_data(),
+                                                             nnz_count);
+        auto row_offsets_event = dal::backend::copy_host2usm(queue,
+                                                             copied_row_offsets.get_mutable_data(),
+                                                             row_offsets_.get_data(),
+                                                             row_count_ + 1);
+        sycl::event::wait_and_throw({ data_event, col_indices_event, row_offsets_event });
+        return csr_table::wrap(copied_data,
+                               copied_col_indices,
+                               copied_row_offsets,
+                               column_count_,
+                               indexing_);
+    }
+#endif // ONEDAL_DATA_PARALLEL
+
+    csr_table build_csr_table(host_test_policy& policy) const {
```

Comment:
Seems like the policy is not used in this function.

Reply:
It is aligned with CPU and GPU tests.
```void csr_general_checks(const te::csr_table_builder& data, bs::result_option_id compute_mode) {
        const auto desc =
            bs::descriptor<float_t, basic_statistics::method::sparse>{}.set_result_options(
                compute_mode);
        const auto csr_table = **data.build_csr_table(this->get_policy());**
        const auto dense_table = data.build_dense_table();
        auto compute_result = this->compute(desc, csr_table);
        table weights;
        check_compute_result(compute_mode, dense_table, weights, compute_result);
    }
```